initial_setup.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# this script performs the initial setup
+
+URL="http://0.0.0.0:8000/"
+status_code=000
+ENV_EXAMPLE=".env.example"
+ENV_ORIG=".env"
+TIMEOUT_TIME=60
+
+error_check() {
+    if [ $? -ne 0 ]; then
+        echo "FAILED"
+        exit
+    fi
+}
+
+# instantiate .env file
+cp ${ENV_EXAMPLE} ${ENV_ORIG}
+
+# runs docker compose as daemon process
+docker-compose up -d
+error_check
+
+# loop until http://0.0.0.0:8000/ is not active and
+# timeout does not occurs
+while [[ ${status_code} -ne 302 && ${SECONDS} -lt ${TIMEOUT_TIME} ]];
+do
+    status_code=$(curl --write-out %{http_code} --silent --output /dev/null ${URL})
+done
+
+# if timeout occurs
+if [[ ${SECONDS} -ge ${TIMEOUT_TIME} ]]; then
+    echo "TIMEOUT OCCURRED"
+    exit
+fi
+
+# run migrations
+docker exec mathesar_service python manage.py migrate
+error_check
```

Comment:
These error/panic checks don't provide any feedback about which command failed. What I often do is something like `cmd_to_run || do_this_in_case_of_failure "this parameter can provide debugging information"`.

Reply:
To provide more feedback to a user, it might be good to print notices about each action being performed as they're being performed. That would also take care of the "where did this panic happen" problem.

Reply:
These error checks don't hide the errors. It's just the way of exiting the script if there's an error so that further lines won't execute.

Reply:
I realise that. My concern is about not immediately knowing which script-line/command failed.